docs/spec/BeakerSpec.tex

```diff
@@ -0,0 +1,1111 @@
+the same as the first $s$ bits of procedure key $b$.
+
+\begin{table}[H]
+  \caption{Delete procedure capability format.}
+  \centering{}%
+  \begin{tabularx}{\textwidth}{c|c|X}
+    \hline
+    Key Offset & Byte Offset & Description \\
+    \hline
+    \hline
+    \texttt{0x00} & \texttt{0x00} & The prefix, which is in the interval [0,24]. \\
+    \texttt{0x00} & \texttt{0x01 - 0x07} & Unsed and undefined. \\
+    \texttt{0x00} & \texttt{0x08 - 0x1f} & The 24 bytes of the base procedure key. \\
+    \hline
+    End &   \\
+    \hline
+  \end{tabularx}
+\end{table}
+
+\includegraphics[width=\textwidth]{media/ProcedureKeyBytes.pdf}
+
+\subsubsection{Capability Subsets}
+As defined in Section~\ref{cap-subsets}, each capability has subsets and
+supersets. For the delete procedure capability a subset is capability which
+possesses the ability to delete no more procedures than the original.
+
+The prefix size of a capability $C$ is given as $C_s$ and the base procedure key
+$C_b$. Given a delete procedure capability $A$ and a proposed subset $B$:
+\begin{itemize}
+  \item If the prefix size of $B$ is equal to or greater than $A$ (that is: $B_s
+  \geq A_s$), and
+  \item If the first $B_s$ bits of $B_b$ is equal to the first $A_s$ bits of
+  $A_b$.
+\end{itemize}
+
+\subsection{Set Entry Procedure System Call}
+The value of the Entry Procedure is 24-byte procedure key which identified which
+procedure should be first called upon receiving a transaction. This is currently
+store at the storage location:
+
+\texttt{0x00 00 00 00 {\color{blue} 04} 00 $\cdots$ 00 00 00 00}
+
+\subsubsection{Error Codes}
+If the Delete Procedure system call fails for any reason other than those
+general system call failure conditions, the system call will return
+\texttt{SYSCALL\_FAIL} (as definted in Section~\ref{errors}) followed by one of
+the following codes.
+
+\begin{table}[H]
+  \caption{Return and error codes.}
+  \centering{}%
+  \begin{tabularx}{\textwidth}{c|c|l|X}
+    \hline
+    Type & Value & Additional Data & Description \\
+    \hline
+    \hline
+    DEL\_NOPROC  & 33 & None & The procedure key specified does not exit. \\
+    \hline
+  \end{tabularx}
+\end{table}
+
+\marginpar[Non-Normative]{\textbf{Non-Normative}}For example, if the Delete
+Procedure system call was executed with a procedure key that does not exist then
+the DELEGATECALL will leave a 0 on the stack and return two bytes
+(\texttt{0x6633}).
+
+\subsubsection{System Call Format}
+
+Type Value: 6
+
+Data: min(64 bytes) max(64 bytes)
+
+\begin{table}[H]
+  \caption{Set entry procedure system call format.}
+  \centering{}%
+  \begin{tabularx}{\textwidth}{c|X}
+    \hline
+    Byte Offset & Description\\
+    \hline
+    \hline
+    \texttt{0x00} & Procedure key; 24 bytes \\
+    \hline
+    End &  \\
+    \hline
+  \end{tabularx}
+\end{table}
+
+\subsubsection{Capability Format}
+This capability has no associated data and is therefore of size zero.
```

Comment:
This is a bit hard to understand, is it a boolean? Or am I misunderstanding something?

Reply:
I've attempted to clarify.

Reply:
So it's a 32 byte `boolean` value, correct?

Reply:
No there is not value. If you think of a capability being type+data, then what would it mean to have a capability of type Set Entry + false? It would be a capability to do nothing. 

In a sense this is the most theoretically pure capability. You possess a Set Entry capability, you can set the entry capability. There is on additional data associated with it. Wherever we have "Capability Format" we're talking about it's associated data. Storage ranges and the like.

Reply:
Ok, I see so all we do is check for length of entry-caps in the c-list.

Reply:
Exactly. *Presence* of a capability is determined by the length of the c-list. The capabilities themselves do not need to carry any data.